### ISSUE TITLE
Firefox Workaround to cross domain issues

### ISSUE THREAD

**Issue by bkolo** [MailOnline/videojs-vast-vpaid#46]
I am familiar with the cross domain issues with firefox.
I brought it to the attention of one of our partners and they gave me an example of their html5 player working in firefox and I thought I would share it with your team.

The main difference i see is that their player does not make an OPTION request to their server before making the GET request for the vast ad.   Your plugin seems to (in firefox) make an OPTION request.  Their server does set the headers correctly on GET requests, but the OPTIONS request is what is failing.  Is there a way to avoid the OPTIONS request?  Perhaps by setting the content-type to XML or some other means?   

Here is an example:
http://dev.testing.optimatic.com/optivpaid/examples/lkqd/

The test ad tag in question is:
http://dev.testing.optimatic.com/optivpaid/examples/vpaid/outputXML.ashx?f=optimatic-vpaid.xml&cb=1444926557

regards,
Bob Kolo

**Comment by carpasse** [comment]
@bkolo I am not sure what you mean with he OPTIONS request. 
Regarding CORS, We do the request withCredentials, is that what you mean an option to decide whether to do the ad request with or without credentials???

**Comment by carpasse** [comment]
@bkolo The OPTIONS request got me confused at first I thought you were talking about a request header.

The Http Options request seems to be the way firefox handles CORS. We don't programatically do the OPTIONS request. It is firefox who triggers it because the domain is different. 

I have found a good blog post regarding the OPTIONS request http://metajack.im/2010/01/19/crossdomain-ajax-for-xmpp-http-binding-made-easy/

If you take a look at the specs http://www.w3.org/TR/cors/ on section "7.1.5 Cross-Origin Request with Preflight" they recommend the use of GET or OPTIONS methods for the preflight request

In case you wonder about the preflight request here is a good explanation http://stackoverflow.com/questions/15381105/cors-what-is-the-motivation-behind-introducing-preflight-requests

Taking another look at mozilla's documentation on CORS they do specify that if you do a request to a different domain, the will do a preflight request.

**In a nutshell, The OPTIONS request is not done on purpose by the plugin but by firefox as a security measure whenever we do a request to a different domain.**

I have not found any way of disabling the preflight OPTIONS request in Firefox. And to be honest after the investigation, I think your partner should be able to handle it.

**Comment by carpasse** [comment]
@bkolo Can you please reach your partner with our answer and tell us their response?

**Comment by bkolo** [comment]
Thank Carlos for the response.
I will close this ticket as I just realized that the ad and the player are
in fact on the same domain which is why the OPTIONS request is not
happening.

regards,
Bob
